XenAdmin/Wizards/CrossPoolMigrateWizard/Filters/CrossPoolMigrateCanMigrateFilter.cs

```diff
@@ -108,103 +100,77 @@ public override bool FailureFound
                                                   vm.opaque_ref,
                                                   receiveMapping,
                                                   true,
-                                                  GetVdiMap(vm, targetSR),
+                                                  GetVdiMap(vm, targetSrs),
                                                   vm.Connection == host.Connection ? new Dictionary<XenRef<VIF>, XenRef<XenAPI.Network>>() : GetVifMap(vm, targetNetwork),
                                                   new Dictionary<string, string>());
                         }
                         catch (Failure failure)
                         {
-                            failureReasons.Add(new FailureReason{Reason = failure.Message, Host = host, Vm = vm});
+                            disableReason = failure.Message;
+                            
+                            log.ErrorFormat("VM: {0}, Host: {1} - Reason: {2};", vm.opaque_ref, host.opaque_ref, failure.Message);
+
+                            if (!excludedHosts.Contains(host.opaque_ref))
+                                excludedHosts.Add(host.opaque_ref);
                         }
                     }
                 }
 
-                return DetermineIfFailureGivenReasons();
+                return excludedHosts.Count == targets.Count;
             }
         }
 
-        /// <summary>
-        /// If we have a pool, then we need to fail only when there are failures for 
-        /// each of the hosts. Otherwise any failure counts.
-        /// </summary>
-        /// <returns></returns>
-        private bool DetermineIfFailureGivenReasons()
+        private List<Host> CollateHosts(out Pool thePool)

```

Comment:
Maybe personal preferences don't match completely here, but of course I am fine with leaving this as is.